src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -46,6 +46,11 @@ internal abstract class AnalyzerDriver : IDisposable
         private Task _primaryTask;
 
         /// <summary>
+        /// Number of worker tasks processing compilation events and executing analyzer actions.
+        /// </summary>
+        private readonly int _workerCount = Math.Min(Environment.ProcessorCount, 10);
```

Comment:
Yes, unless someone hands me a machine with 64 cores, I can't verify the performance profile though :) I'll remove the bound.